compiler/src/dotty/tools/dotc/core/tasty/TreeUnpickler.scala

```diff
@@ -331,6 +333,8 @@ class TreeUnpickler(reader: TastyReader, nameAtRef: NameRef => TermName, posUnpi
           ConstantType(Constant(readType()))
         case ENUMconst =>
           ConstantType(Constant(readTermRef().termSymbol))
+        case SCALASYMBOLconst =>
```

Comment:
I don't think you need the duplication. SYMBOLconsts are simple types, so they should just occur in readSimpleType. I believe we duplication of CLASSconst and ENUMconst is also wrong and should be removed.